packages/gradient_boosting_model/gradient_boosting_model/processing/preprocessors.py

```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class SklearnTransformerWrapper(BaseEstimator, TransformerMixin):
+    """Categorical data missing value imputer."""
+
+    def __init__(self, variables=None, transformer=None):
+
+        if not isinstance(variables, list):
+            self.variables = [variables]
+        else:
+            self.variables = variables
+
+        self.transformer = transformer
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit statement to accomodate the sklearn pipeline."""
+        self.transformer.fit(X[self.variables])
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """Apply the transforms to the dataframe."""
+        X = X.copy()
+        X[self.variables] = self.transformer.transform(X[self.variables])
+        return X
+
+
+class TemporalVariableEstimator(BaseEstimator, TransformerMixin):
+    """Temporal variable calculator."""
+
+    def __init__(self, variables=None, reference_variable=None):
+        if not isinstance(variables, list):
+            self.variables = [variables]
+        else:
+            self.variables = variables
+
+        self.reference_variables = reference_variable
+
+    def fit(self, X, y=None):
+        # we need this step to fit the sklearn pipeline
+        return self
+
+    def transform(self, X):
+        X = X.copy()
+        for feature in self.variables:
+            X[feature] = X[self.reference_variables] - X[feature]
+
+        return X
+
+
+class DropUnecessaryFeatures(BaseEstimator, TransformerMixin):
+    def __init__(self, variables_to_drop=None):
+        self.variables = variables_to_drop
+
+    def fit(self, X, y=None):
+        return self
```

Comment:
Here I would add:
"fit() method is necessary to accomodate the Scikit-learn pipeline functionality"

Reply:
updated